tests/bson/bson-binary-clone-001.phpt

```diff
@@ -18,7 +18,6 @@ $types = array(
 foreach($types as $type) {
     // Use 16-byte data to satisfy UUID requirements
     $binary = new MongoDB\BSON\Binary('randomBinaryData', $type);
-    $binary->foo = 'bar';
```

Comment:
This was actually a regression test for [PHPC-1230](https://jira.mongodb.org/browse/PHPC-1230) dating back to https://github.com/mongodb/mongo-php-driver/commit/aed838497bd8f87fb2a5b0cc0fbf0e66a8a6d54d.

If we're removing this across the board, would it make sense to reintroduce a `clone` test for at least one class that runs on PHP versions before 8.2?

Reply:
That's a good point - I'll duplicate the tests and have them create dynamic properties on PHP before 8.2. 👍

Reply:
Duplicated each test for PHP 8.2 and restored the originals 👍

Reply:
If this test is now restricted to PHP < 8.2, shouldn't we add back the dynamic property testing? This might have been an oversight.

Also, quoting [`version_compare`](https://www.php.net/version_compare):

> Note that pre-release versions, such as 5.3.0-dev, are considered lower than their final release counterparts (like 5.3.0).

So I think we do want to use `<?php skip_if_php_version('>=', '8.1.99'); ?>` as discussed in https://github.com/mongodb/mongo-php-driver/pull/1340#discussion_r934725681